resources/examples/manager/BasicAssetLibrary/plugin/BasicAssetLibrary/BasicAssetLibraryInterface.py

```diff
@@ -123,11 +130,13 @@ def resolve(self, entityRefs, traitSet, context, hostSession):
                     trait_data = entity.traits.get(trait)
                     if trait_data is not None:
                         self.__add_trait_to_traits_data(trait, trait_data, result)
-            except Exception as exc:  # pylint: disable=broad-except
-                exc_name = exc.__class__.__name__
-                result = EntityResolutionError(f"{exc_name}: {exc}", ref)
-            results.append(result)
-        return results
+
+                successCb(idx, result)
+            except KeyError:
```

Comment:
[fix] Are we sure that only `KeyErrors` are relevant here? There could be other errors from `TraitsData.setTraitProperty` or unpacking to `namedtuple` depending on what is in the JSON, which might be nice to keep as batch errors vs breaking the whole call? if we do want to make those exceptions rather than tie them to the ref (which is helpful), we should document the reason for only catching `KeyError` and leaving the rest.

Reply:
Hmm. I think if there's any other error than a `KeyError` from `bal.entity(...)`, then it's a programmer error and should be exceptional?

How about a dedicated exception type for "unknown entity" to replace `KeyError`?

Reply:
Added a dedicated exception type in 847d58e8, then restructured a bit in 068d8645 and 7f554ad7

Reply:
Based on offline conversation, did a little refactor in c4bf164d to allow unexpected `KeyError`s to propagate.

Reply:
This is great @feltech a much better approach, capturing our off-line chat, as this is effectively calling out a subtle but important part of the resolve API contract, can we make sure that we have something in the documentation for `resolve` in `Manager`/`ManagerInterface` that makes the HTTP analogy - that `5xx` should always be an exception, and `4xx` batch result error?

Reply:
Is it sufficient for that analogy to be in the docs for `BatchElementError` as added in https://github.com/OpenAssetIO/OpenAssetIO/pull/569#discussion_r946580578 ?

Reply:
I think It'd be nice to have some spell-it-out-like-I'm-five examples in the documentation for `resolve` directly, so that someone implementing the method in a manager has a nice cheat sheet to hand, when they're learning about the obligations of that method. For the host - making it really obvious what the failure modes are, and what they mean in terms of business logic would be really useful.

Reply:
I've expanded the docs a little in 13e8925f, how is that?

Reply:
Perfect, thanks!